rendermodules/pointmatch/generate_point_matches_opencv.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+import json
+import renderapi
+import cv2
+import multiprocessing
+from six.moves import urllib
+import logging
+from argschema import ArgSchemaParser
+from .schemas import \
+        PointMatchOpenCVParameters, \
+        PointMatchClientOutputSchema
+
+
+example = {
+        "ndiv": 8,
+        "matchMax": 10000,
+        "downsample_scale": 0.3,
+        "SIFT_nfeature": 50000,
+        "SIFT_noctave": 8,
+        "SIFT_sigma": 2.5,
+        "RANSAC_outlier": 5.0,
+        "FLANN_ntree": 5,
+        "FLANN_ncheck": 50,
+        "ratio_of_dist": 0.8,
+        "CLAHE_grid": 16,
+        "CLAHE_clip": 2.5,
+        "pairJson": "/allen/programs/celltypes/production/wijem/workflow_data/production/reference_2018_06_21_21_44_56_00_00/jobs/job_27418/tasks/task_25846/tile_pairs_em_2d_raw_lc_stack_z_2908_to_2908_dist_0.json",
+        "input_stack": "em_2d_raw_lc_stack",
+        "match_collection": "dev_collection_can_delete",
+        "render": {
+              "owner": "TEM",
+              "project": "em_2d_montage_staging",
+              "host": "em-131db",
+              "port": 8080,
+              "client_scripts": "/allen/aibs/pipeline/image_processing/volume_assembly/render-jars/production/scripts"
+            },
+        "ncpus": 7
+        }
+
+
+def ransac_chunk(fargs):
+    [k1xy, k2xy, des1, des2, k1ind, args] = fargs
+
+    FLANN_INDEX_KDTREE = 0
+    index_params = dict(
+            algorithm=FLANN_INDEX_KDTREE,
+            trees=args['FLANN_ntree'])
+    search_params = dict(checks=args['FLANN_ncheck'])
+    flann = cv2.FlannBasedMatcher(index_params, search_params)
+    MIN_MATCH_COUNT = 10
+
+    matches = flann.knnMatch(des1[k1ind, :], des2, k=2)
+
+    # store all the good matches as per Lowe's ratio test.
+    good = []
+    k1 = []
+    k2 = []
+    for m, n in matches:
+        if m.distance < args['ratio_of_dist']*n.distance:
+            good.append(m)
+    if len(good) > MIN_MATCH_COUNT:
+        src_pts = np.float32(
+                [k1xy[k1ind, :][m.queryIdx] for m in good]).reshape(-1, 1, 2)
+        dst_pts = np.float32(
+                [k2xy[m.trainIdx] for m in good]).reshape(-1, 1, 2)
+        M, mask = cv2.findHomography(
```

Comment:
For the sake of utility, it could be nice to include a multiple hypothesis RANSAC option like Stephan has implemented in Java.  We can have that as a separate PR as well.

Reply:
I'm not familiar with that feature at the moment, so, I'd agree with separate PR.